src/Akka.Streams.Kafka.Tests/Integration/RebalanceExtTests.cs

```diff
@@ -0,0 +1,381 @@
+        var tpFutureMap = new Dictionary<string, TaskCompletionSource<Done>>();
+        foreach (var i in Enumerable.Range(1, topicCount))
+        {
+            var topic1 = CreateTopic(1);
+            await GivenInitializedTopicAsync(topic1, partitionCount);
+            Log.Debug("Created topic {0}", topic1);
+            topics.Add(topic1);
+            foreach (var j in Enumerable.Range(0, partitionCount))
+            {
+                tps.Add(new TopicPartition(topic1, j));
+                tpFutureMap[$"{topic1}-{j}"] = new TaskCompletionSource<Done>();
+            }
+        }
+
+        var (producerTpsAck, messageAndStoreAck) =
+            PublishMessages(partitionCount, perPartitionMessageCount, topics);
+
+        return new TopicPartitionMetadata(topics, tps, tpFutureMap, producerTpsAck, messageAndStoreAck);
+    }
+
+    private
+        (List<Task<Done>> producerTpsAck, Dictionary<int, MessageAck> messageAndStoreAck)
+        PublishMessages(int partitionCount, int perPartitionMessageCount, IReadOnlyList<string> topics)
+    {
+        var messageAndStoreAck = new Dictionary<int, MessageAck>();
+        var producerTpsAck = new List<Task<Done>>();
+        var topicIndex = -1;
+        foreach (var topic in topics)
+        {
+            topicIndex += 1;
+            var topicOffset = topicIndex * partitionCount * perPartitionMessageCount;
+            foreach (var partitionIdx in Enumerable.Range(0, partitionCount))
+            {
+                var startMessageIndx = partitionIdx * perPartitionMessageCount + 1 + topicOffset;
+                var endMessageIndx = startMessageIndx + perPartitionMessageCount - 1;
+                var messageRange = Enumerable.Range(startMessageIndx, perPartitionMessageCount).ToList();
+                var topicPartition = new TopicPartition(topic, new Partition(partitionIdx));
+                var partitionName = $"{topic}-{partitionIdx}";
+                foreach (var messageId in messageRange)
+                {
+                    messageAndStoreAck[messageId] = new MessageAck(partitionName, new AtomicCounter(0),
+                        new TaskCompletionSource<Done>(), new TaskCompletionSource<Done>());
+                }
+
+                producerTpsAck.Add(DoProduce());
+                continue;
+
+                async Task<Done> DoProduce()
+                {
+                    await ProduceStrings(topicPartition, messageRange, ProducerSettings);
+                    Log.Debug("publishMessages:published messages from ({0} to {1}) for partition name {2}",
+                        startMessageIndx, endMessageIndx,
+                        partitionName);
+                    return Done.Instance;
+                }
+            }
+        }
+
+        return (producerTpsAck, messageAndStoreAck);
+    }
+
+    [Fact(DisplayName =
+        "Fetched records must not be lost when two consumers consume from one topic and two partitions and one consumer aborts mid-stream")]
```

Comment:
This is a very complicated spec that I ported from https://github.com/akka/alpakka-kafka/releases/tag/v3.0.1 - it doesn't translate 1:1 with .NET because we don't have fine-grained control over partition assignments, but it does assert that we don't lose messages even if one partitioned consumer gets aborted mid-flight.